model/Storage.inc.php

```diff
@@ -575,6 +575,28 @@ public static function updateFileItemInfo($item, $storageFileID, Zotero_StorageF
 		}
 		$item->save();
 		
+		self::addFileLibraryReference($storageFileID, $item->libraryID);
+		
+		Zotero_DB::commit();
+	}
+	
+	public static function addFileLibraryReference($storageFileID, $libraryID) {
+		$sql = "INSERT IGNORE INTO storageFileLibraries (storageFileID, libraryID) VALUES (?,?)";
+		return Zotero_DB::query($sql, array($storageFileID, $libraryID));
+	}
+	
+	public static function deleteFileLibraryReference($storageFileID, $libraryID) {
+		Zotero_DB::beginTransaction();
+		
+		$sql = "SELECT COUNT(*) FROM storageFileItems WHERE storageFileID = ?";
```

Comment:
Doesn't matter in this case, but `SELECT 1 FROM storageFileItems WHERE storageFileID = ? LIMIT 1` is a better pattern to use here. (Elsewhere we use `SELECT COUNT(*) > 0, but it looks like that's not actually optimized in current MySQL versions.) And then change variable below to `$exists` (or just put `valueQuery` in the conditional).